src/snowflake/cli/_plugins/connection/util.py

```diff
@@ -54,6 +50,53 @@ def __init__(self, host: str | None):
         )
 
 
+class UIParameter(Enum):
+    ENABLE_REGIONLESS_REDIRECT = "UI_SNOWSIGHT_ENABLE_REGIONLESS_REDIRECT"
+    EVENT_SHARING_V2 = "ENABLE_EVENT_SHARING_V2_IN_THE_SAME_ACCOUNT"
+    ENFORCE_MANDATORY_FILTERS = (
+        "ENFORCE_MANDATORY_FILTERS_FOR_SAME_ACCOUNT_INSTALLATION"
+    )
+
+
+def get_ui_parameter(
+    conn: SnowflakeConnection, parameter: UIParameter, default: Any
+) -> str:
+    """
+    Returns the value of a single UI parameter.
+    If the parameter is not found, the default value is returned.
+    """
+
+    if not isinstance(parameter, UIParameter):
```

Comment:
Is the reason for having this check explicitly along with the type hint just to be extra safe?

Reply:
yes

Reply:
we could make that argument about any function. we need to decide if we trust our tools or not (in this case mypy). i think this is safe enough without, especially since this function is in the code path for many integ tests